packages/webview_flutter/webview_flutter_platform_interface/lib/src/v4/webview_controller_delegate.dart

```diff
@@ -0,0 +1,264 @@
+    throw UnimplementedError(
+        'WebView goBack is not implemented on the current platform');
+  }
+
+  /// Goes forward in the history of this WebView.
+  ///
+  /// If there is no forward history item this is a no-op.
+  Future<void> goForward() {
+    throw UnimplementedError(
+        'WebView goForward is not implemented on the current platform');
+  }
+
+  /// Reloads the current URL.
+  Future<void> reload() {
+    throw UnimplementedError(
+        'WebView reload is not implemented on the current platform');
+  }
+
+  /// Clears all caches used by the [WebView].
+  ///
+  /// The following caches are cleared:
+  ///	1. Browser HTTP Cache.
+  ///	2. [Cache API](https://developers.google.com/web/fundamentals/instant-and-offline/web-storage/cache-api) caches.
+  ///    These are not yet supported in iOS WkWebView. Service workers tend to use this cache.
+  ///	3. Application cache.
+  ///	4. Local Storage.
+  Future<void> clearCache() {
+    throw UnimplementedError(
+        'WebView clearCache is not implemented on the current platform');
+  }
+
+  /// Evaluates a JavaScript expression in the context of the current page.
+  ///
+  /// The Future completes with an error if a JavaScript error occurred, or if the type of the
+  /// evaluated expression is not supported (e.g on iOS not all non-primitive types can be evaluated).
+  Future<String> evaluateJavaScript(String javascript) {
+    throw UnimplementedError(
+        'WebView evaluateJavaScript is not implemented on the current platform');
+  }
+
+  /// Runs the given JavaScript in the context of the current page.
+  ///
+  /// The Future completes with an error if a JavaScript error occurred.
+  Future<void> runJavaScript(String javascript) {
+    throw UnimplementedError(
+        'WebView runJavaScript is not implemented on the current platform');
+  }
+
+  /// Runs the given JavaScript in the context of the current page, and returns the result.
+  ///
+  /// The Future completes with an error if a JavaScript error occurred, or if the
+  /// type the given expression evaluates to is unsupported. Unsupported values include
+  /// certain non-primitive types on iOS, as well as `undefined` or `null` on iOS 14+.
+  Future<String> runJavaScriptReturningResult(String javascript) {
+    throw UnimplementedError(
+        'WebView runJavaScriptReturningResult is not implemented on the current platform');
+  }
+
+  /// Adds new JavaScript channels to the set of enabled channels.
+  ///
+  /// For each value in this list the platform's webview should make sure that a corresponding
+  /// property with a postMessage method is set on `window`. For example for a JavaScript channel
+  /// named `Foo` it should be possible for JavaScript code executing in the webview to do
+  ///
+  /// ```javascript
+  /// Foo.postMessage('hello');
+  /// ```
+  ///
+  /// See also: [CreationParams.javascriptChannelNames].
+  Future<void> addJavaScriptChannels(Set<String> javascriptChannelNames) {
```

Comment:
Users will be calling this method directly, so they will need to pass the callback method as well. I was thinking something like

```dart
Future<void> addJavaScriptChannel(AddJavaScriptChannelParams params) {
    throw UnimplementedError();
}

class AddJavaScriptChannelParams {
  AddJavaScriptChannelParams({
    required this.name,
    required this.onMessageReceived,
  });

  final String name;

  final void Function(JavaScriptMessage) onMessageReceived;
}
```

And then `removeJavaScriptChannels` can be changed to

```dart
Future<void> removeJavaScriptChannel(String name) {
  throw UnimplementedError();
}
```

Reply:
That makes the API a lot nicer, updated the code to reflect this in commit 8684ac3.